jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/assembly/AssemblyManagerTest.java

```diff
@@ -48,6 +55,7 @@ public class AssemblyManagerTest {
   public void setUp() throws Exception {
     assemblyManager = AssemblyManager.getInstance();
     targetDirectory = temporaryFolder.newFolder("target");
+    logger = mock(KitLogger.class);
```

Comment:
Why did you choose to initialize `logger` instead of using `@Mock` annotation? Was it not working with the annotation?

ohk, it was regarding https://github.com/eclipse/jkube/pull/1617#discussion_r904732898 . You need to do this :
```suggestion
    logger = new KitLogger.SilentLogger();
```

Reply:
As discussed we can't use SilentLogger for verifications, Hence writing this way. Please verify now @rohanKanojia

Reply:
As discussed we can't use SilentLogger for verification. @rohanKanojia Please verify now.

Reply:
We can't use `mock` but we can use `spy` to observe interactions with object:

```java
logger = spy(new KitLogger.SilentLogger());
```

Reply:
What Rohan mention would be the consistent approach, we should be shifting to this in all of our tests, please replace if you can.